File: pingyougou_parent/pingyougou_manager_web/src/main/webapp/js/controller/BrandController.js
app.controller("brandController",function ($scope,$http,$controller,brandService) {
    $controller('baseController',{$scope:$scope}) //继承
    $scope.findAll=function () {
        $http.get('../brand/findAll.do').success(function (response) {
            $scope.list=response;
        })
    }

    /*无条件分页查询*/
    $scope.findPage = function (page,pageSize) {
        brandService.findPage(page,pageSize).success(function (response) {
            $scope.list=response.rows;
            $scope.paginationConf.totalItems=response.total;
        })
    }
    //增加
    $scope.save=function () {
        var Object=brandService.add($scope.entity);
        if($scope.entity.id != null){
            Object = brandService.update($scope.entity)
        }
        Object.success(function (response) {
            if(response.success){
                $scope.reLoad();
            }else{
                alert(response.message);
            }
        })
    }

    //按id查询
    $scope.findOne= function (id) {
        brandService.findOne(id).success(function (response) {
            $scope.entity=response;
        })
    }



    //删除方法
    $scope.dele=function () {
        brandService.dele($scope.delIds).success(function (response) {
            if(response.success){
                $scope.reLoad();
            }else{
                alert(response.message);
            }
        })
    }
    //自定义查询对象
    $scope.searchEntity={};
    //查询
    $scope.search=function (page,pageSize) {
        brandService.search(page,pageSize,$scope.searchEntity).success(function (response) {
            $scope.paginationConf.totalItems=response.total;
            $scope.list=response.rows;
        })
    }
})

File: pingyougou_parent/pingyougou_manager_web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.Brand;
import com.pinyougou.sellergoods.service.BrandService;
import entity.PageResult;
import entity.Result;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

/*
 *-----------------神兽保佑 -----------------
 *           ┌─┐        ┌─┐
 *       ┌─┘  ┴────┘  ┴─┐
 *       │                        │
 *       │        ───          │
 *       │  ─┬┘       └┬─   │
 *       │                        │
 *       │       ─┴─           │
 *       │                        │
 *       └───┐          ┌──┘
 *               │          │
 *               │          │
 *               │          │
 *               │          └────────────┐
 *               │                                    │
 *               │                                    ├─┐
 *               │                                    ┌─┘
 *               │                                    │
 *               └─┐  ┐  ┌─────┬─┐  ┌──┘
 *                   │  ┤  ┤          │  ┤  ┤
 *                   └─┴─┘          └─┴─┘
 *
 * ------------------代码无BUG!--------------
 * @Author: Seesky
 * @Date: 2018/8/20 16:57
 * @Description:
 */
@RestController
@RequestMapping("/brand")
public class BrandController {

    @Reference
    private BrandService brandService;

    @RequestMapping("/findAll")
    public List<Brand> findAll(){
        return brandService.findAll();
    }

    @RequestMapping("/add")
    public Result add(@RequestBody Brand brand){
        try {
            brandService.add(brand);
            return new Result(true,"保存成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"保存失败");
        }
    }

    @RequestMapping("/delete")
    public Result delete(Long[] ids){
        try {
            brandService.delete(ids);
            return new Result(true,"删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"删除失败");
        }
    }

    @RequestMapping("/findOne")
    public Brand findOne(Long id){

        return brandService.findOne(id);
    }
    @RequestMapping("/findPage")
    public PageResult findPage(int page,int pageSize){
        return brandService.fingPage(page, pageSize);
    }

    @RequestMapping("/search")
    public PageResult search(@RequestBody Brand brand,int page ,int pageSize){
        return brandService.findPage(page, pageSize, brand);
    }

    @RequestMapping("/update")
    public Result update(@RequestBody Brand brand){
        try {
            brandService.update(brand);
            return new Result(true,"修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"修改失败");
        }
    }

    @RequestMapping("/getBrandList")
    public List<Map> getBrandList(){
        return brandService.getBrandList();
    }
}
